builtin/checkout.c

```diff
@@ -101,6 +101,8 @@ struct branch_info {
 	char *checkout;
 };
 
+struct repository *r;
```

Comment:
Better move this into the `merge_working_tree()` function. Oh, and don't forget to assign it: `struct repository *r = the_repository;`